Social-tap-API/Social-tap-API/Calculations.cs

```diff
@@ -0,0 +1,103 @@
+            barName = char.ToUpper(barName[0]) + barName.Substring(1);
+
+            return barName;
+        }
+        /// Į metodą paduodamas baro pavadinimas ir jo įvertinimas
+        /// Web service laikome baro pavadinimą ir jo įvertinimų Listą 
+        /// Atgal grąžiname tik baro įvertinimų vidurkį
+        /// http://localhost:.../api/values/barrate/string_baropavadinimas/string_įvertinimas
+        public double BarRateAverage(string barName, int rate)
+        {
+
+            if (_barRates.Keys.Contains(barName))
+            {
+                _barRates[barName].Add(rate);
+            }
+            else
+            {
+                _barRates[barName] = new List<int> { rate };
+            }
+            return _barRates[barName].Average();
+        }
+
+        /// Į metodą paduodama informacija kiek buvo įpilta 
+        /// Web service saugoma informacija apie tai kiek kartų buvo pasinaudota programa 
+        /// Grąžiname true jeigu įpilta geriau, false jeigu blogau. 
+        /// Lyginimas vyksta ne su vieno baro statistika o su BENDRA 
+        /// http://localhost:.../api/values/bevlvl/INT
+       // [HttpPost("bevlvl/{beverageLevel}")]    
+        public Boolean Average(int beverageLevel)
+        {
+            _uses++;
+            _sum += beverageLevel;
+            // sakysime, kad jeigu lygus vidurkiui, 
+            // tai ipilta geriau
+            return _sum / _uses <= beverageLevel;
+        }
+
+        /// Į metodą paduodamas komentaras 
+        /// Iš jo išrenkami žodžiai panaudoti su hashtagu 
+        /// Grąžinamas hashtagų listas 
+        /// http://localhost:.../api/values/tags/STRING
+        //  [HttpPost("tags/{comment}")]
+        public List<string> HashtagsFinder(string comment)
+        {
+            /*
+             * Hashtag'ą programoje reikės pakeist kuo nors kitu naudojant kintamasis.
+             * Replace("#","Ę"), ir tada passinti į web API, kol kas dedu Ę nes neturėtų
+             * būti naudojamas komentaruose kaip pirma raidė sakinio. Bet galima sugalvoti kuo kitu keisti.
+             */
+            var regex = new Regex(@"(?<=Ę)\w+");
+            var matches = regex.Matches(comment);
+
+            foreach (Match m in matches)
+            {
+                _hashTags.Add(m.Value);
+
+            }
+            return _hashTags;
+        }
+        /// <summary>
+        /// Patikrina ar paduoti tinkami kintamieji
+        /// Ar rate mažiau lygu 5 ir daugiau nei 0
+        /// Ar beverage mažiau lygu 10 ir daugiau nei 0 
+        /// Ar baro vardas yra bent 1 simblos 
+        public Boolean Validation(int rate, int beverage, string barName)
+        {
+            if (rate > MAX_RATE || beverage > MAX_BEVERAGE_LEVEL || barName.Length < MIN_NAME_LENGHT || rate <= MIN_BEVERAGE_RATE_LEVEL || beverage <= MIN_BEVERAGE_RATE_LEVEL)
```

Comment:
Šitą galima daug švariau parašyt. `return <sąlyga>` vietoj 
`if (<sąlyga>)
   return true;
else 
  return false;`. Visiškai tas pats, lengviau skaitosi.

Reply:
Sutvarkyta